app/dao/notifications_dao.py

```diff
@@ -249,7 +253,8 @@ def update_notification_status_by_id(
 @transactional
 def update_notification_status_by_reference(reference, status):
     # this is used to update letters and emails
-    notification = Notification.query.filter(Notification.reference == reference).first()
+    stmt = select(Notification).where(Notification.reference == reference)
+    notification = db.session.execute(stmt).scalar_one_or_none()
 
     if not notification:
```

Comment:
I changed this to `if notification is None`.  In this case, that is clearly correct.